app/Http/Controllers/ModelController.php

```diff
@@ -2,62 +2,66 @@
 
 namespace App\Http\Controllers;
 
-use Illuminate\Http\Request;
 use App\Models\Brand;
+use App\Repository\ModelRepositoryInterface;
+use Illuminate\Http\Request;
 
 class ModelController extends Controller
 {
-    public function model()
+
+    private ModelRepositoryInterface $modelRepository;
+
+    public function __construct(ModelRepositoryInterface $modelRepository)
     {
-        $models = Brand::all();
-        return view('model', ['models' => $models]);
+        $this->modelRepository = $modelRepository;
     }
 
+    public function model()
+    {
+        $model = $this->modelRepository->all();
+        return view('Model', ['models' => $model]);
+    }
 
-
-    public function show($id)
+    public function show($modelId)
     {
-    
-        $data = Brand::find($id);
         return response()->json([
+            'data' => $this->modelRepository->findModel($modelId),
             'status' => 'success',
             'message' => 'exito',
-            'data' => $data
         ]);
+
     }
-    
+
     public function store(Request $request)
     {
-        $data = Brand::create($request->all());
+        $data = $this->modelRepository->createModel($request->all());
         return response()->json([
```

Comment:
Investiga sobre los Resources: https://laravel.com/docs/11.x/eloquent-resources
En lugar de tener:
```
return response()->json([
            'status' => 'success',
            'message' => 'exito',
            'data' => $data
            ]);
```
Tendrías únicamente:
`return response()->json($this->modelRepository->createModel($request->all()));`

De igual forma, investiga sobre las validaciones (Request Validation) y aplícalas en tus controladores: https://laravel.com/docs/11.x/validation#form-request-validation

En lugar de tener:
`public function update(Request $request, $id)`

Ahora tendrías: 
public function update(ModelUpdateRequest $request)

Reply:
De este estoy teniendo uno pequeños detalles pero eso ando investigando